feat: NFC in proximity and rp test plans references in normative texts and further test plans where necessary (#832)
italia/eidas-it-wallet-docs: docs/en/remote-flow.rst

Remote Flow
===========

> Depending on whether the User is using a mobile device or a workstation, the Relying Party MUST support the following remote flows (:ref:`test-plans-remote-presentation:RPR-01`, :ref:`test-plans-remote-presentation:RPR-02`, :ref:`test-plans-remote-presentation:RPR-03`, :ref:`test-plans-remote-presentation:RPR-04`, :ref:`test-plans-remote-presentation:RPR-05`, :ref:`test-plans-remote-presentation:RPR-06`):

Author: ```suggestion
Depending on whether the User is using a mobile device or a workstation, the Relying Party MUST support the following remote flows (:ref:`test-plans-remote-presentation:RPR-01`, :ref:`test-plans-remote-presentation:RPR-02`):
```

> * **Same Device**, the Relying Party MUST provide an HTTP location to the Wallet Instance using a redirect (302) or an HTML href in a web page (:ref:`test-plans-remote-presentation:RPR-01`, :ref:`test-plans-remote-presentation:RPR-02`);

Author: ```suggestion
* **Same Device**, the Relying Party MUST provide an HTTP location to the Wallet Instance using a redirect (302) or an HTML href in a web page (:ref:`test-plans-remote-presentation:RPR-01`);
```

> * **Cross Device**, the Relying Party MUST provide a QR Code which the User frames with the Wallet Instance (:ref:`test-plans-remote-presentation:RPR-03`, :ref:`test-plans-remote-presentation:RPR-04`).

Author: ```suggestion
* **Cross Device**, the Relying Party MUST provide a QR Code which the User frames with the Wallet Instance (:ref:`test-plans-remote-presentation:RPR-02`).
```

Remote Flow
===========

Depending on whether the User is using a mobile device or a workstation, the Relying Party MUST support the following remote flows (:ref:`RPR-97 <test-plans-remote-presentation:RPR-97>`):

* **Same Device**, the Relying Party MUST provide an ``HTTP`` location to the Wallet Instance using a redirect (``302``) or an HTML href in a web page (:ref:`RPR-01 <test-plans-remote-presentation:RPR-01>`);
* **Cross Device**, the Relying Party MUST provide a ``QR Code`` which the User frames with the Wallet Instance (:ref:`RPR-03 <test-plans-remote-presentation:RPR-03>`).

> Once the ``Wallet Instance`` establishes the trust with the ``Relying Party`` and evaluates the request, the User gives the consent for the disclosure of the Digital Credentials, in the form of a ``Verifiable Presentation``.

Author: ```suggestion
Once the Wallet Instance establishes the trust with the Relying Party and evaluates the request, the User gives the consent for the disclosure of the Digital Credentials, in the form of a Verifiable Presentation.
```

**Steps 11-13 (Request URI Request)**: The Wallet Instance checks if the Relying Party has provided the ``request_uri_method`` within its signed Request Object.

>   - If it is provided and is equal to post, the Wallet Instance SHOULD provide its metadata to the Relying Party. The Relying Party updates the Request Object according with the Wallet technical capabilities.

Author: ```suggestion
  - If it is provided and is equal to ``post``, the Wallet Instance SHOULD provide its metadata to the Relying Party. The Relying Party updates the Request Object according with the Wallet technical capabilities.
```

  
      response=eyJhbGciOiJFUzI1NiIs...9t2LQ
  

>   Below is a non-normative example of the decrypted payload of the JWT contained in the response, before base64url encoding. The vp_token parameter value corresponds to the format used when the DCQL query language is used in the presentation request.

Author: ```suggestion
  Below is a non-normative example of the decrypted payload of the JWT contained in the response, before base64url encoding. The ``vp_token`` parameter value corresponds to the format used when the DCQL query language is used in the presentation request.
```

      HOST: relying-party.example.org

  When the Wallet Instance has provided the presentation to the Relying Party's **response_uri** endpoint and the User authentication is successful. The Relying Party updates the session cookie allowing the user-agent to access to the protected resource. A redirect URL is provided carrying the location where the user-agent is intended to navigate.

>   The following is a non-normative example of the response with the redirect_uri from the Relying Party to the user-agent.

Author: ```suggestion
  The following is a non-normative example of the response with the ``redirect_uri`` from the Relying Party to the user-agent.
```

.. warning::
  The Wallet Instance, when providing its technical capabilities to the
  Relying Party, MUST NOT include any User information or other explicit (:ref:`RPR-100 <test-plans-remote-presentation:RPR-100>`)
  information regarding the hardware used or usage preferences of its User (:ref:`RPR-92 <test-plans-remote-presentation:RPR-92>`).

> If both the Relying Party and the Wallet Instance support the ``request_uri_method`` with HTTP POST, the Wallet Instance capabilities (metadata) MUST be provided using an HTTP request to the `request_uri` endpoint of the Relying Party, with the method POST and content type set to `application/x-www-form-urlencoded` (:ref:`RPR-101 <test-plans-remote-presentation:RPR-101>`).

Author: ```suggestion
If both the Relying Party and the Wallet Instance support the ``request_uri_method`` with HTTP POST, the Wallet Instance capabilities (metadata) MUST be provided using an HTTP request to the ``request_uri`` endpoint of the Relying Party, with the method POST and content type set to ``application/x-www-form-urlencoded`` (:ref:`RPR-101 <test-plans-remote-presentation:RPR-101>`).
```

   * - `authorization_endpoint`
     - URL of the authorization server's endpoint, see `OAUTH2`_. Using an universal link is preferable for enhanced security and fallback support, custom url schemes can also be used if necessary.
   * - `response_types_supported`

>      - OPTIONAL. JSON array of OAuth 2.0 "response_type" values. If present it MUST be set to `vp_token` (:ref:`RPR-102 <test-plans-remote-presentation:RPR-102>`).

Author: ```suggestion
     - OPTIONAL. JSON array of OAuth 2.0 ``response_type`` values. If present it MUST be set to ``vp_token`` (:ref:`RPR-102 <test-plans-remote-presentation:RPR-102>`).
```

>      - OPTIONAL. JSON array of OAuth 2.0 "response_type" values. If present it MUST be set to `vp_token` (:ref:`RPR-102 <test-plans-remote-presentation:RPR-102>`). Default is `vp_token`.

Author: ```suggestion
     - OPTIONAL. JSON array of OAuth 2.0 "response_type" values. If present it MUST be set to `vp_token` (:ref:`RPR-102 <test-plans-remote-presentation:RPR-102>`).
```
